### ISSUE TITLE
All changes are lost in project creation when following links with keyboard 

### ISSUE THREAD

**Issue by sashathomas** [ManifoldScholar/manifold#2629]
Hi! This is Sasha Thomas. I am a new accessibility student working for Hadi Rangin at UW. I'm looking forward to helping make Manifold more accessible.

In Manifold, when creating a project, it is important that users are warned when unsaved changes are about to get deleted. When selecting a link with a keyboard, the "unsaved changes" modal flashes briefly, and then follows the link, and gets rid of any changes made to the project. 

This issue occurs on Chrome and Firefox (Mac and PC), and on Safari. 

## Expected Behavior

All users should be be made aware of unsaved changes that are about to get deleted when following a link. Whether you are navigating with a mouse or a keyboard, the "unsaved changes" modal should appear and warn users that their changes will be deleted upon leaving the page.

## Current Behavior

When selecting links with a keyboard, the "unsaved changes" modal flashes, and the new page is loaded immediately. This will automatically delete any changes a keyboard user has made to the project without their consent. 

## Possible Solution

The solution is to make sure the page is not left with unsaved changes, unless the user intentionally says they want to. 

## Steps to Reproduce

1. Navigate to admin interface (through Enter Admin Mode)
2. Navigate to an editable project
3. Under general attributes, make any change to the current settings (title, subtitle, etc.)
4. Navigate to the left sidebar using keyboard
5. Follow any of the links on the left sidebar by hitting enter
6. Note that the modal flashes, and the new page is loaded
7. If you navigate back to the general attributes page, notice that none of your changes are saved. 

## Context (Environment)

Mac OS Catalina v10.15.1
Google Chrome Version 80.0.3987.106
Firefox 68 ESR

**Comment by zdavis** [comment]
@bot bug

**Comment by zdavis** [comment]
Thank you for taking the time to report this bug. We've reviewed the issue and agree that this should be fixed. We’re adding a “planned” label to indicate that we consider this bug as part of our current work plan.

_This was an automated message, but please don't hesitate to reply. Our team watches these issues closely and will respond as soon as we're able to!_

**Comment by sashathomas** [comment]
I'm getting similar behavior in the "texts" tab in the admin panel. When I attempt to delete to a previously uploaded text (such as a Google Doc, .epub, etc.) using a mouse, a similar modal appears confirming my decision to delete the text. However, when using a keyboard, this modal does not appear, and the text is deleted immediately. 

### Steps to Reproduce
1. Create a project and upload any piece of text to it, such as a .epub file.
2. Using a mouse, navigate to the "delete" button (visually a trashcan) and click the link.
3. Note that there is a modal confirming that you want to delete this text.
4. Click no, navigate to the "delete" button again using keyboard, and hit enter.
5. Note that there was no confirmation modal, but the text has been deleted.

**Comment by zdavis** [comment]
Yep, I'm able to reproduce this. We'll fix this for v5 RC2.
